lti_consumer/plugin/views.py

```diff
@@ -501,3 +556,430 @@ def list(self, *args, **kwargs):
+    else:
+        return JsonResponse(
+            {'error': 'The provided session_data claim does not match the anti-CSRF token.'},
+            status=HTTP_400_BAD_REQUEST,
+        )
+
+    # TODO: The resource link should uniquely represent the assessment in the Assessment Platform.
+    #       We SHOULD provide a value for the title attribute.
+    #       It's RECOMMENDED to provide a value for the description attribute.
+    #       The xblock-lti-consumer library does not currently support setting these attributes.
+    resource_link = request.POST.get('resource_link')
+    attempt_number = request.POST.get('attempt_number')
+
+    lti_consumer.set_proctoring_data(
+        attempt_number=attempt_number,
+        session_data=session_data,
+        resource_link=resource_link,
+    )
+
+    # TODO: I hardcoded this to None for right now. We'll need to figure out how to supply user_roles from outside the
+    #       platform.
+    # TODO: LTI Proctoring Services expects that the user role is empty or includes
+    #       "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner", which is a context role.
+    #       lti_consumer.set_user_data uses the regular roles. So, for now, I'm leaving it empty.
+    # Required user claim data
+    lti_consumer.set_user_data(
+        user_id=request.user.id,
+        role=None,
+    )
+
+    # These claims are optional. They are necessary to set in order to properly verify the verified_user claim,
+    # if the Proctoring Tool includes it in the JWT.
+    # TODO: This will need to have additional consideration for PII.
+    # optional_user_identity_claims = get_optional_user_identity_claims()
+    # lti_consumer.set_proctoring_user_data(
+    #     **optional_user_identity_claims
+    # )
+
+    try:
+        lti_response = lti_consumer.check_and_decode_token(token)
+    # TODO: This needs better error handling.
+    except (BadJwtSignature, MalformedJwtToken, NoSuitableKeys):
+        return JsonResponse(
+            {},
+            status=HTTP_400_BAD_REQUEST,
+        )
+
+    # If the Proctoring Tool specifies the end_assessment_return claim in its LTI launch request,
+    # the Assessment Platform MUST send an End Assessment Message at the end of the user's
+    # proctored exam.
+    end_assessment_return = lti_response.get('end_assessment_return')
+    if end_assessment_return:
+        end_assessment_return_key = get_cache_key(app="lti", key="end_assessment_return", user_id=request.user.id)
+        # We convert the boolean to an int because memcached will return an int even if a boolean is stored. This
+        # ensures a consistent return value.
+        end_assessment_return_value = int(end_assessment_return)
+        TieredCache.set_all_tiers(end_assessment_return_key, end_assessment_return_value)
+
+    return JsonResponse(data={})
+
+
+# We do not want Django's CSRF protection enabled for POSTs made by external services to this endpoint. This is because
```

Comment:
I don't think that django's CSRF not being able to handle this case is worth an ADR. It might be worth separating this note out into some readme if we have a suitable one (but I don't think we do)

I honestly would have just written "django's CSRF can't handle a flow where it isn't in control of the middle" so this is already more info than I demand :)